test/prompt/test_prompt_node.py

```diff
@@ -724,6 +741,37 @@ def test_simple_pipeline_yaml(tmp_path):
     assert result["results"][0] == "positive"
 
 
+@pytest.mark.integration
+def test_simple_pipeline_yaml_with_customized_params(tmp_path):
+    with open(tmp_path / "tmp_config.yml", "w") as tmp_file:
+        tmp_file.write(
+            """
+            version: ignore
+            components:
+            - name: Prompter
+              type: PromptNode
+            pipelines:
+            - name: query
+              nodes:
+              - name: Prompter
+                inputs:
+                - Query
+        """
+        )
+    the_question = "What is the capital of Germany?"
+    pipeline = Pipeline.load_from_yaml(path=tmp_path / "tmp_config.yml")
+
+    with patch.object(HFLocalInvocationLayer, "invoke", MagicMock()) as mock_call:
```

Comment:
Yes, something like this mocking approach should work for all layers.

Reply:
the problem with all other layers is that they require `api_key`.